### DISCUSSION TITLE
how to write auto close autocmd in lua nvim_create_autocmd api?

### DISCUSSION THREAD

**Discussion by nshen** [nvim-tree/nvim-tree.lua#1130] [Q&A] [answered]
`autocmd BufEnter * ++nested if winnr('$') == 1 && bufname() == 'NvimTree_' . tabpagenr() | quit | endif`

how to write the command ?

```lua
vim.api.nvim_create_autocmd(
   "BufEnter",
   { command = "++nested if winnr('$') == 1 && bufname() == 'NvimTree_' . tabpagenr() | quit | endif" }
 )
```

**Comment by nshen** [comment]
I have tried a lot. Is this correct?

```lua
vim.api.nvim_create_autocmd("BufEnter", {
  callback = function()
    if vim.fn.winnr("$") == 1 and vim.fn.bufname() == ("NvimTree_" .. vim.fn.tabpagenr()) then
      vim.cmd("quit")
    end
  end,
})
```

**Reply by nshen** [reply]
But I don't understand what `++nested` mean.

**Reply by gegoune** [reply]
`:h autocmd-nested` describes it, it is so events get executed for commands/events in your top level autocommand.

**Reply by nshen** [reply]
> `:h autocmd-nested` describes it, it is so events get executed for commands/events in your top level autocommand.

oh thanks I thought it was some variable increasing. 

final version:

```lua
vim.api.nvim_create_autocmd("BufEnter", {
  nested = true,
  callback = function()
    if vim.fn.winnr("$") == 1 and vim.fn.bufname() == ("NvimTree_" .. vim.fn.tabpagenr()) then
      vim.cmd("quit")
    end
  end,
})
```

**Comment by kyazdani42** [comment, marked as answer]
check the existing discussions please... https://github.com/kyazdani42/nvim-tree.lua/discussions/1115

**Reply by nshen** [reply]
oh thanks, I think your answer  should be `~= nil`
